### ISSUE TITLE
json_name only supports character strings

### ISSUE THREAD

**Issue by hckhilliam** [stephenh/ts-proto#496]
When compiling a protobuf that has a `json_name` param, the name must be a valid variable name format, or else it will throw an exception like the following example:

Message
```
message Foo {
  string bar = 1 [json_name="bar:baz"]
}
```

Exception
```
error TS1005: ',' expected.
18692 version: isSet(object.bar:baz)
```

I assume the fix should instead use `object['bar:baz']` instead of the dot notation

**Comment by stephenh** [comment]
@hckhilliam shoot, that's a good point.

I wonder we could have `jsonName` be conditionally `['foo:bar']` if we detect an invalid symbol, around in here:

https://github.com/stephenh/ts-proto/blob/main/src/main.ts#L1295 

Hopefully that would "just work" without too many changes to the rest of that code, and ideally without changing the current output for valid symbols.

I haven't looked at the from json side of things.

If you're interested in working on a PR, that'd be great! Thanks!

**Comment by stephenh** [comment]
This is fixed by #520 , thanks @boukeversteegh !
